Mathlib/RingTheory/Artinian.lean

```diff
@@ -498,4 +498,26 @@ instance isMaximal_of_isPrime (p : Ideal R) [p.IsPrime] : p.IsMaximal :=
       localization_surjective (nonZeroDivisors (R ⧸ p)) (FractionRing (R ⧸ p))⟩).isField _
     <| Field.toIsField <| FractionRing (R ⧸ p)
 
+variable (R) in
+lemma prime_spectrum_finite : {I : Ideal R | I.IsPrime}.Finite := by
+  let Spec := {I : Ideal R | I.IsPrime}
+  obtain ⟨_, ⟨s, rfl⟩, H⟩ := set_has_minimal
+    (range <| (Finset.inf · (fun x ↦ x.1) : Finset Spec → Ideal R)) ⟨⊤, ⟨∅, by simp⟩⟩
+  refine ⟨⟨s, fun p ↦ ?_⟩⟩
+  obtain ⟨q, ⟨hq1, hq2⟩⟩ : ∃ q ∈ s, q ≤ p
+  · classical
+    specialize H (p ⊓ s.inf fun x ↦ x.1) ⟨insert p s, by simp⟩
```

Comment:
Does this not work?
```suggestion
    specialize H p ⟨{p}, by simp⟩
```

Reply:
I think its easier to use `p ⊓ ⨅ s`, because it is guaranteed to be smaller than or equal to the right hand side, so more lemmas can be applied